src/app.controller.ts

```diff
@@ -23,38 +23,46 @@ export class AppController {
 
   @Post('createNewGame')
   @Redirect('/')
-  async createNewGame(@Body() gameParams: { size: number }) {
-    const newGameId = (await this.appService.createNewGame(gameParams.size)).id;
+  async createNewGame(@Body() gameParams: { size: number }, @Req() req: Request) {
+    const newGameId = (await this.appService.createNewGame(gameParams.size, req.sessionID)).id;
     return { url: `/game/${newGameId}` };
   }
 
+  @Get('game/:id/join')
+  @Redirect('/')
+  async joinGame(@Param('id') id: number, @Req() req: Request) {
+    const gameId = (await this.appService.joinGame(id, req.sessionID)).id;
```

Comment:
I find it weird that you use the return value while you know the game id from the request. It makes me think that there is a hidden logic in appService.joinGame, but there is not...